src/Parlot/Scanner.cs

```diff
@@ -114,8 +108,10 @@ public bool ReadFirstThenOthers(Func<char, bool> first, Func<char, bool> other,
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    [Obsolete("Should not be part of the Scanner")]
     public bool ReadIdentifier() => ReadIdentifier(out _);
 
+    [Obsolete("Should not be part of the Scanner")]

```

Comment:
What should be used instead? A note on that would be useful here.

Reply:
The parsers already expose Identifier methods.

Reply:
I don't understand how that would help here: https://github.com/OrchardCMS/OrchardCore/blob/0b7b4a335e861d50a80f1280fab81f26b0d9d9fb/src/OrchardCore.Modules/OrchardCore.Shortcodes/Controllers/AdminController.cs#L351 Could you please tell?

Reply:
Yeah, that's a weird thing to do. This should be provided by the shortcode library. To fix this I would suggest to build an actual shortcode out of the provided name and try to parse it "successfully". My sarcastic mind is glad it exposed bad usages of this ;)